spec/betagouvbot/webhook_spec.rb

```diff
@@ -0,0 +1,23 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+RSpec.describe BetaGouvBot::Webhook do
+  describe '/compte' do
+    let(:callback)  { 'https://bob.coop' }
+    let(:user_name) { 'bob' }
+    let(:text)      { "#{user_name} #{user_name}@email.coop password" }
+    let(:params)    { { response_url: callback, user_name: user_name, text: text } }
+
+    before do
+      stub_request(:any, callback)
+      stub_request(:any, 'https://beta.gouv.fr/api/v1.3/authors.json')
+        .to_return(
+          headers: { 'content-type' => 'application/json; charset=utf-8' },
+          body: [id: user_name].to_json
+        )
+    end
+
+    it { expect(post('/compte')).not_to be_ok }
+    it { expect(post('/compte', params)).to be_ok }
```

Comment:
Unless I'm missing something, this should return a status of OK in both cases; what varies is the response supplied to the callback URL.

From the perspective of a Slack user, a status different from OK will only result in a vague and uninformative error message; to be more informative, we need to go to the callback URL and POST a response.